google/cloud/aiplatform/base.py

```diff
@@ -561,6 +579,123 @@ def _sync_object_with_future_result(
             if value:
                 setattr(self, attribute, value)
 
+    def _construct_sdk_resource_from_gapic(
+        self,
+        gapic_resource: proto.Message,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> AiPlatformResourceNoun:
+        """Given a GAPIC object, return the SDK representation."""
+        sdk_resource = self._empty_constructor(credentials=credentials)
+        sdk_resource._gca_resource = gapic_resource
+        return sdk_resource
+
+    # TODO(b/144545165) - Improve documentation for list filtering once available
+    @classmethod
+    def list(
+        cls,
+        filter: Optional[str] = None,
+        order_by: Optional[str] = None,
+        page_size: Optional[int] = None,
+        read_mask: Optional[field_mask.FieldMask] = None,
+    ) -> Sequence[AiPlatformResourceNoun]:
+        """List all instances of this AI Platform Resource.
+
+        Example Usage:
+
+        aiplatform.BatchPredictionJobs.list(
+            filter='state="JOB_STATE_SUCCEEDED" AND display_name="my_job"',
+        )
+
+        aiplatform.Model.list(order_by="create_time desc, display_name")
+
+        Args:
+            filter (str):
+                Optional. An expression for filtering the results of the request.
+                For field names both snake_case and camelCase are supported.
+            order_by (str):
+                Optional. A comma-separated list of fields to order by, sorted in
+                ascending order. Use "desc" after a field name for descending.
+                Supported fields: `display_name`, `create_time`, `update_time`
+            page_size (int):
+                Optional. The standard list page size.
+            read_mask (field_mask.FieldMask):
+                Optional. Mask specifying which fields to read.
+
+        Returns:
+            Sequence[AiPlatformResourceNoun] - A list of SDK resource objects
+        """
+        _UNSUPPORTED_LIST_ORDER_BY_TYPES = (
+            aiplatform.jobs._Job,
+            aiplatform.models.Endpoint,
+            aiplatform.models.Model,
+            aiplatform.training_jobs._TrainingJob,
+        )
+
+        self = cls._empty_constructor()
+
+        creds = initializer.global_config.credentials
+
+        resource_list_method = getattr(self.api_client, self._list_method)
+        order_locally = False
+
+        list_request = {
+            "parent": initializer.global_config.common_location_path(),
+            "filter": filter,
+            "page_size": page_size,
+            "read_mask": read_mask,
+        }
+
+        # If list method does not offer `order_by` field, order locally
+        if order_by and issubclass(type(self), _UNSUPPORTED_LIST_ORDER_BY_TYPES):
+            order_locally = True
+        elif order_by:
+            list_request["order_by"] = order_by
+
+        resource_list = resource_list_method(request=list_request) or []
```

Comment:
Why is the empty list needed here?

Reply:
I ran into a few rare instances where the service returned a `None` instead of an empty list, this just serves as a fail safe against that.